src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -387,9 +383,16 @@ def plan_state(self) -> Dict[PlanState, Any]:
                     num_selects_with_complexity_merged += 1
             height += 1
             current_level = next_level
+        # calculate the repeated node status
+        cte_nodes, duplicated_node_complexity_distribution = find_duplicate_subtrees(
+            self, propagate_complexity_hist=True
+        )
+
         return {
             PlanState.PLAN_HEIGHT: height,
             PlanState.NUM_SELECTS_WITH_COMPLEXITY_MERGED: num_selects_with_complexity_merged,
+            PlanState.NUM_CTE_NODES: len(cte_nodes),
```

Comment:
it's the same as "num_duplicate_nodes" before, right? I'm not sure whether we should change the name since it will break the existing query in dashboard

Reply:
There is no change to the telemetry node, it is just a change for our internal code to be more clear at least from our internal developer's point of view. If you look at the telemetry sent here https://github.com/snowflakedb/snowpark-python/pull/2494/files#diff-c46aa591b19bb02d4b80a69dda3c5b4ff9e2ff4ad18cf1170779aa678ccbdafaR194 it is still using the same name